### ISSUE TITLE
Unable to reboot or poweroff from initrd emergency shell

### ISSUE THREAD

**Issue by RomanValov** [systemd/systemd#35757]
### systemd version the issue has been seen with

257.1

### Used distribution

Debian testing

### Linux kernel version used

6.12.6-amd64

### CPU architectures issue was seen on

x86_64

### Component

systemctl

### Expected behaviour you didn't see

System goes into reboot or poweroff when using appropriate command

### Unexpected behaviour you saw

Invocation of reboot/poweroff ends up with error and nothing happens.

FYI: when using pre-257 systemd versions: the error message is shown anyway, nonetheless reboot/poweroff is performed

### Steps to reproduce the problem

1) Build initrd image using dracut and reboot
2) Boot into emergency shell using either `rd.emergency` or `rd.break` kernel command line parameter
3) Run `reboot` or `poweroff`

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
Generating "/run/initramfs/rdsosreport.txt"


Entering emergency mode. Exit the shell to continue.
Type "journalctl" to view system logs.
You might want to save "/run/initramfs/rdsosreport.txt" to a USB stick or /boot
after mounting them and attach it to a bug report.


Press Enter for maintenance
(or press Control-D to continue): 
:/root# reboot
Bus n/a: changing state UNSET → OPENING
sd-bus: starting bus by connecting to /run/dbus/system_bus_socket...
Bus n/a: changing state OPENING → CLOSED
Failed to connect to system scope bus via local transport: No such file or directory
:/root# systemctl --version
systemd 257 (257.1-4)
```

**Comment by RomanValov** [comment]
Symptoms looks the same as with #35416 but in different context. Commands works fine in rescue mode on my system

**Comment by bluca** [comment]
debug logs would be useful, please try with `SYSTEMD_LOG_LEVEL=debug systemctl reboot` instead and report back the output

**Comment by RomanValov** [comment]
Hm. Sorry for misleading example in bug report. The output was indeed collected with debug log level. For clarity:

```
:/root# SYSTEMD_LOG_LEVEL=debug systemctl reboot
Bus n/a: changing state UNSET → OPENING
sd-bus: starting bus by connecting to /run/dbus/system_bus_socket...
Bus n/a: changing state OPENING → CLOSED
Failed to connect to system scope bus via local transport: No such file or directory
```

Also for clarity: when I have mentioned rescue mode above, I meant rescue mode invoked via `systemctl rescue` from booted system (as per #35416). However when using rescue mode on boot (`rd.rescue`) the problem is reproduced.

**Comment by yuwata** [comment]
Fix is waiting in #35773.
